fitbenchmarking/results_processing/support_page.py

```diff
@@ -13,86 +13,84 @@
+    :param result: The result for a specific benchmark problem-minimizer-etc
+                   combination
+    :type result: fitbenchmarking.utils.fitbm_result.FittingResult
     :param support_pages_dir: directory to store the support pages in
     :type support_pages_dir: str
     :param options: The options used in the fitting problem and plotting
     :type options: fitbenchmarking.utils.options.Options
     """
-
-    for result in prob_results:
-        prob_name = result.sanitised_name
-
-        file_name = f"{prob_name}_{result.sanitised_min_name}.html"
-        file_name = file_name.lower()
-        file_path = os.path.join(support_pages_dir, file_name)
-
-        # Bool for print message/insert image
-        fit_success = init_success = options.make_plots
-
-        if options.make_plots:
-            fig_fit, fig_start = get_figure_paths(result)
-            if fig_fit == '':
-                fig_fit = result.figure_error
-                fit_success = False
-            if fig_start == '':
-                fig_start = result.figure_error
-                init_success = False
-        else:
-            fig_fit = fig_start = 'Re-run with make_plots set to yes in the ' \
-                                  'ini file to generate plots.'
-
-        root = os.path.dirname(inspect.getfile(fitbenchmarking))
-        template_dir = os.path.join(root, "templates")
-        env = Environment(loader=FileSystemLoader(template_dir))
-        css = get_css(options, support_pages_dir)
-        template = env.get_template("support_page_template.html")
-
-        with open(file_path, 'w') as fh:
-            fh.write(template.render(
-                css_style_sheet=css['main'],
-                table_style=css['table'],
-                custom_style=css['custom'],
-                title=result.name,
-                equation=result.problem.equation,
-                initial_guess=result.ini_function_params,
-                minimizer=result.minimizer,
-                is_best_fit=result.is_best_fit,
-                initial_plot_available=init_success,
-                initial_plot=fig_start,
-                min_params=result.fin_function_params,
-                fitted_plot_available=fit_success,
-                fitted_plot=fig_fit))
-
-        result.support_page_link = os.path.relpath(file_path)
+    prob_name = result.sanitised_name
+
+    file_name = f'{prob_name}_{result.costfun_tag}_' \
+                f'{result.sanitised_min_name}.html'
+    file_name = file_name.lower()
+    file_path = os.path.join(support_pages_dir, file_name)
+
+    # Bool for print message/insert image
+    fit_success = init_success = options.make_plots
+
+    if options.make_plots:
+        fig_fit, fig_start = get_figure_paths(result)
+        if fig_fit == '':
+            fig_fit = result.figure_error
+            fit_success = False
+        if fig_start == '':
+            fig_start = result.figure_error
+            init_success = False
```

Comment:
```suggestion
        fit_success, init_success = fig_fit != "", fig_start != ""
        fig_fit = result.figure_error if not fit_success else fig_fit
        fig_start = result.figure_error if not init_success else fig_start
```

Reply:
I'll make this a bit neater, but I think it's a bit more readable with an if on a seperate line when one of the assignments is a noop.